volumina/widgets/exportHelper.py

```diff
@@ -35,63 +34,110 @@
         raise RuntimeError(
             "can not export from a non-lazyflow data source (layer=%r, datasource=%r)"
             % (type(layer), type(layer.datasources[0]))
         )
+    return [
+        slot.dataSlot
+        for (slot, isSlot) in zip(layer.datasources, sourceTags)
+        if isSlot is True
+    ]
 
-    if not _has_lazyflow:
-        raise RuntimeError("lazyflow not installed")
-    import lazyflow
 
-    dataSlots = [slot.dataSlot for (slot, isSlot) in zip(layer.datasources, sourceTags) if isSlot is True]
+def _get_stacked_data_sources(layer: Layer) -> OpMultiArrayStacker:
+    """Get operator with stacked output from all slots of layer
+
+    Args:
+        layer (Layer): layer containing datasources that can be exported
+          (datasource.dataSlot -> Slot)
+
+    Returns:
+        OpMultiArrayStacker: Operator configured with all stackable input slots
+        from layer.
+    """
+    dataSlots = _get_export_slots(layer)
 
-    opStackChannels = lazyflow.operators.OpMultiArrayStacker(dataSlots[0].getRealOperator().parent)
+    opStackChannels = lazyflow.operators.OpMultiArrayStacker(
+        dataSlots[0].getRealOperator().parent
+    )
     for slot in dataSlots:
-        assert isinstance(slot, lazyflow.graph.Slot), "slot is of type %r" % (type(slot))
-        assert isinstance(slot.getRealOperator(), lazyflow.graph.Operator), "slot's operator is of type %r" % (
-            type(slot.getRealOperator())
+        assert isinstance(slot, lazyflow.graph.Slot), "slot is of type %r" % (
+            type(slot)
         )
+        assert isinstance(
+            slot.getRealOperator(), lazyflow.graph.Operator
+        ), "slot's operator is of type %r" % (type(slot.getRealOperator()))
     opStackChannels.AxisFlag.setValue("c")
     opStackChannels.Images.resize(len(dataSlots))
     for i, islot in enumerate(opStackChannels.Images):
         islot.connect(dataSlots[i])
 
-    export_dir = PreferencesManager().get("layer", "export-dir", default=os.path.expanduser("~"))
+    return opStackChannels
 
+
+def get_export_operator(layer: Layer) -> OpFormattedDataExport:
+    """Get export operator configured with stacked output from layer
+
+    Args:
+        layer (Layer): layer containing datasources that can be exported
+          (datasource.dataSlot -> Slot)
+
+    Returns:
+        OpFormattedDataExport: Operator configured with all stackable input slots
+        from layer.
+    """
+    opStackChannels = _get_stacked_data_sources(layer)
     # Create an operator to do the work
-    from lazyflow.operators.ioOperators import OpFormattedDataExport
 
     opExport = OpFormattedDataExport(parent=opStackChannels.parent)
-    opExport.OutputFilenameFormat.setValue(os.path.join(export_dir, layer.name))
     opExport.Input.connect(opStackChannels.Output)
     opExport.TransactionSlot.setValue(True)
 
+    return opExport
+
+
+def get_settings_and_export_layer(layer: Layer, parent_widget=None) -> None:
```

Comment:
The `get` prefix suggests that this function should return something. Consider renaming to `prompt_layer_export_settings` or something similar.

Reply:
good point. I went for `prompt_export_settings_and_export_layer`, not very handy but at least descriptive.